### DISCUSSION TITLE
What should learn when develop my required extension?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [microsoft/vscode-discussions#262] [Extension Development QnA] [answered]
I have an idea of a extension:

```
# ==start. Function1 ==
def func1() {
...
}
# ==end. Function: func1 ==
```

```
// ==start. Function: func1 ==
function func1() {
...
}
// ==end. Function: func1 ==
```

1. I want to there is a panel for me to create the `Function: func1` and `Property: address` items, then I can use them when I wrap my code.
2. I want to when I type `// ==`, there will hint the options for me to select. 
3. I want to the code which is wrapped by my defined annotations(such as `Function/Property/Macro-Variable/` ) has it's own custom background color. 

please tell me what I should learn in VSCode Extension development.

**Comment by isidorn** [comment, marked as answer]
Start here https://code.visualstudio.com/api
